includes/class-jetpack.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Jetpack integration class.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main class.
+ */
+class Jetpack {
+
+	/**
+	 * Modules scripts handles.
+	 *
+	 * @var string[]
+	 */
+	private static $scripts_handles = [
+		'jp-tracks',              // Tracks analytics library.
+		'jetpack-instant-search', // Jetpack Instant Search.
+		'jetpack-search-widget',  // Jetpack Search widget.
+	];
+
+	/**
+	 * Initialize hooks and filters.
+	 */
+	public static function init() {
+		add_action( 'wp_enqueue_scripts', [ __CLASS__, 'jetpack_async_scripts' ], 20 );
+		add_filter( 'newspack_amp_plus_sanitized', [ __CLASS__, 'jetpack_modules_amp_plus' ], 10, 2 );
+	}
+
+
+	/**
+	 * Make Jetpack scripts async.
+	 */
+	public static function jetpack_async_scripts() {
+		foreach ( self::$scripts_handles as $handle ) {
+			wp_script_add_data( $handle, 'async', true );
+		}
+	}
+
+	/**
+	 * Allow Jetpack modules scripts to be loaded in AMP Plus mode.
+	 *
+	 * @param bool|null $is_sanitized If null, the error will be handled. If false, rejected.
+	 * @param object    $error        The AMP sanitisation error.
+	 *
+	 * @return bool Whether the error should be rejected.
+	 */
+	public static function jetpack_modules_amp_plus( $is_sanitized, $error ) {
+		if ( isset( $error, $error['node_attributes'], $error['node_attributes']['id'] ) ) {
+			$has_any_id = array_reduce(
```

Comment:
Nit: `$has_any_id` kinda threw me off – I'd expect something like `$script_has_matching_id`.

Reply:
Agreed! Updated both matching variables in 13da562ee8f60d11abbbc794853c3c09a4d650c9